keel-core/src/test/kotlin/com/netflix/spinnaker/keel/actuation/IntermittentFailureTests.kt

```diff
@@ -0,0 +1,170 @@
+package com.netflix.spinnaker.keel.actuation
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceKind.Companion.parseKind
+import com.netflix.spinnaker.keel.api.actuation.Task
+import com.netflix.spinnaker.keel.api.id
+import com.netflix.spinnaker.keel.api.plugins.ResourceHandler
+import com.netflix.spinnaker.keel.api.plugins.SupportedKind
+import com.netflix.spinnaker.keel.core.api.randomUID
+import com.netflix.spinnaker.keel.diff.DefaultResourceDiff
+import com.netflix.spinnaker.keel.events.ResourceCreated
+import com.netflix.spinnaker.keel.events.ResourceValid
+import com.netflix.spinnaker.keel.pause.ActuationPauser
+import com.netflix.spinnaker.keel.persistence.ArtifactRepository
+import com.netflix.spinnaker.keel.persistence.DeliveryConfigRepository
+import com.netflix.spinnaker.keel.persistence.memory.InMemoryDiffFingerprintRepository
+import com.netflix.spinnaker.keel.persistence.memory.InMemoryResourceRepository
+import com.netflix.spinnaker.keel.persistence.memory.InMemoryUnhappyVetoRepository
+import com.netflix.spinnaker.keel.test.DummyResourceSpec
+import com.netflix.spinnaker.keel.test.deliveryConfig
+import com.netflix.spinnaker.keel.test.resource
+import com.netflix.spinnaker.keel.veto.VetoEnforcer
+import com.netflix.spinnaker.keel.veto.unhappy.UnhappyVeto
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService
+import com.netflix.spinnaker.time.MutableClock
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.mockk
+import java.time.Clock
+import kotlinx.coroutines.runBlocking
+import org.springframework.context.ApplicationEventPublisher
+
+/**
+ * This class tests a specific scenario:
+ * (1) a resource has a diff which we cannot fix, so it is vetoed.
+ * (2) that diff goes away (it was an intermittent failure of another system, maybe) and we say it's happy
+ * (3) that same diff comes back, and we still try again because it went from happy to diff.
```

Comment:
Just so I understand, number (3) here is in opposition to not trying anymore because it stayed unhappy forever?

Reply:
yeah - that's correct.

Reply:
Also, judging from the code in `ResourceActuator`, it shouldn't matter if the diff here is the same as in (1) or a different one, right? If there's any change at all between desired and current, `diff.hasChanges()` will return true and the fingerprint will be stored.

Reply:
yes, correct! however, the same diff is just a more mentally weird case that i'm testing out here